backend/src/nodes/nodes/external_stable_diffusion/img2img.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+import numpy as np
+
+from . import category as ExternalStableDiffusionCategory
+from ...impl.external_stable_diffusion import (
+    decode_base64_image,
+    SamplerName,
+    STABLE_DIFFUSION_IMG2IMG_URL,
+    post,
+    encode_base64_image,
+)
+from ...node_base import NodeBase, group
+from ...node_factory import NodeFactory
+from ...properties.inputs import (
+    TextInput,
+    NumberInput,
+    SliderInput,
+    EnumInput,
+    ImageInput,
+)
+from ...properties.outputs import ImageOutput
+from typing import Optional
+
+
+@NodeFactory.register("chainner:external_stable_diffusion:img2img")
+class Img2Img(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Modify an image using Automatic1111"
+        self.inputs = [
+            ImageInput(),
+            TextInput("Prompt", default="an astronaut riding a horse"),
+            TextInput("Negative Prompt").make_optional(),
+            SliderInput(
+                "Denoising Strength",
+                minimum=0,
+                default=0.75,
+                maximum=1,
+                slider_step=0.01,
+                controls_step=0.1,
+                precision=2,
+            ),
+            group("seed")(
+                NumberInput("Seed", minimum=0, default=42, maximum=4294967296)
+            ),
+            SliderInput("Steps", minimum=1, default=20, maximum=150),
+            EnumInput(SamplerName, default_value=SamplerName.EULER),
+            SliderInput(
+                "CFG Scale",
+                minimum=1,
+                default=7,
+                maximum=20,
+                controls_step=0.1,
+                precision=1,
+            ),
+            SliderInput("Width", minimum=64, default=512, maximum=2048),
+            SliderInput("Height", minimum=64, default=512, maximum=2048),
+            TextInput("Model Checkpoint Override").make_optional(),
+        ]
+        self.outputs = [
+            ImageOutput(),
```

Comment:
We talked about the Width and Height inputs before and how the REST service seems to guarantee that the returned images have those dimensions. If we actually guarantee that the returned images have those dimensions (e.g. by checking and throwing an error if they aren't the right size), then we can denote this in the output type here.

Chainner has a custom type system that allows us to verify chains and to predict output types. E.g. when you see chainner knowing the exact size of all output images in [this screenshot](https://github.com/chaiNNer-org/chaiNNer/blob/main/src/public/screenshot.png), that's the type system at work. 

Assuming that the Width and Height inputs are correct, we can denotes in the `ImageOutput` like this:
```py
ImageOutput(image_type="Image { width: Input8, height: Input9 }")
```

`Input8` and `Input9` are the Width and Height inputs. All inputs are assigned an id based on their position (starting with 0) and those to inputs have the ids 8 and 9 respectively.

But we can potentially do even better. Do we also know the number of channels of the returned image? E.g. if given a grayscale, RGB, or RGBA image, what type of image will be returned? If we do, then we can denote this in the type as well.

Reply:
Very cool!

It looks like `Image to Image` will always return an RGB image of the size you specify, even if it's given a greyscale or RGBA input, or an input with a different size.  `Text to Image` is the same.

I added these notes to the ImageOutputs.